website/docs/r/site_recovery_replication_recovery_plan.html.markdown

```diff
@@ -0,0 +1,148 @@
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_site_recovery_replication_recovery_plan"
+description: |-
+    Manages a Site Recovery Replication Recovery Plan on Azure.
+---
+
+# azurerm_site_recovery_replication_recovery_plan
+
+Manages a Azure Site Recovery Plan within a Recovery Services vault. A recovery plan gathers machines into recovery groups for the purpose of failover.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "source" {
+  name     = "example-source-rg"
+  location = "west us"
+}
+
+
+resource "azurerm_resource_group" "target" {
+  name     = "example-target-rg"
+  location = "east us"
+}
+
+resource "azurerm_recovery_services_vault" "example" {
+  name                = "example-kv"
+  location            = azurerm_resource_group.target.location
+  resource_group_name = azurerm_resource_group.target.name
+  sku                 = "Standard"
+}
+
+resource "azurerm_site_recovery_fabric" "source" {
+  resource_group_name = azurerm_resource_group.example.name
+  recovery_vault_name = azurerm_recovery_services_vault.example.name
+  name                = "example-fabric-source"
+  location            = azurerm_resource_group.source.location
+}
+
+resource "azurerm_site_recovery_fabric" "target" {
+  resource_group_name = azurerm_resource_group.target.name
+  recovery_vault_name = azurerm_recovery_services_vault.example.name
+  name                = "example-fabric-target"
+  location            = azurerm_resource_group.target.location
+  depends_on          = [azurerm_site_recovery_fabric.source]
+}
+
+resource "azurerm_site_recovery_replication_recovery_plan" "example" {
+  name                      = "example-recover-plan"
+  resource_group_name       = azurerm_resource_group.target.name
+  recovery_vault_name       = azurerm_recovery_services_vault.target.name
+  source_recovery_fabric_id = azurerm_site_recovery_fabric.source.id
+  target_recovery_fabric_id = azurerm_site_recovery_fabric.target.id
+
+  recovery_groups {
+    group_type                 = "Boot"
+    replicated_protected_items = [azurerm_site_recovery_replicated_vm.test.id]
+  }
+  recovery_groups {
+    group_type = "Failover"
+  }
+  recovery_groups {
+    group_type = "Shutdown"
+  }
+
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the Replication Plan.
+
+* `resource_group_name` - (Required) Name of the resource group where the vault that should be updated is located.
+
+* `recovery_vault_name` - (Required) The name of the vault that should be updated.
+
+* `source_recovery_fabric_id` - (Required) ID of source fabric to be recovered from. Changing this forces a new Replication Plan to be created.
+
+* `target_recovery_fabric_id` - (Required) ID of target fabric to recover. Changing this forces a new Replication Plan to be created.
+
+* `recovery_group` - (Required) Three or more `recovery_group` block. At least one `recovery_group` for every `group_type` needed.
```

Comment:
Not quite understand the second sentence. `group_type` is one element of `recovery_group`, so each `recovery_group` has a `group_type` ?

Reply:
yes, and there should be at least three `recovery_group` Maybe we can make `recovery_group` a computed one to avoid this?

Reply:
Can it be `computed`? Isn't `recovery_group` a configurable property?

Reply:
I remember there is some trouble as it's `typeSet`.. if I set it to `computed` there might be a diff, let me give it a try.

Reply:
The default hash for `Set` will ignore the `computed` properties. The problem is: if this property is configurable by users, it should not be set as `computed`?

Reply:
it's configurable and if the users do not configure it, it will also be generated by service. 
I have not set it to `computed` and asking users to set at least one group for every type.
My problem is the block, itself is `computed`, while in this situation, it get response with 3 `recovery_group`, but only one in the config file, then there will be a diff.

Reply:
Souds like a backend problem. Normally, the backend should not change the configuration sent by the frontend.